python/dgl/nn/pytorch/explain/subgraphx.py

```diff
@@ -364,3 +365,445 @@ def explain_graph(self, graph, feat, target_class, **kwargs):
+        ----------
+        mcts_node : MCTSNode
+            Node in MCTS
+
+        Returns
+        -------
+        list
+            Children nodes after pruning
+        """
+        if len(mcts_node.children) > 0:
+            return mcts_node.children
+
+        subg = node_subgraph(self.graph, mcts_node.nodes)
+        # Choose k nodes based on the highest degree in the subgraph
+        node_degrees_map = {
+            ntype: torch.zeros(
+                subg.num_nodes(ntype), device=subg.nodes(ntype).device
+            )
+            for ntype in subg.ntypes
+        }
+        for c_etype in subg.canonical_etypes:
+            src_ntype, _, dst_ntype = c_etype
+            node_degrees_map[src_ntype] += subg.out_degrees(etype=c_etype)
+            node_degrees_map[dst_ntype] += subg.in_degrees(etype=c_etype)
+
+        node_degrees_list = [
+            ((ntype, i), degree)
+            for ntype, node_degrees in node_degrees_map.items()
+            for i, degree in enumerate(node_degrees)
+        ]
+        node_degrees = torch.stack([v for _, v in node_degrees_list])
+        k = min(subg.num_nodes(), self.num_child)
+        chosen_node_indicies = torch.topk(
+            node_degrees, k, largest=self.high2low
+        ).indices
+        chosen_nodes = [node_degrees_list[i][0] for i in chosen_node_indicies]
+
+        mcts_children_maps = dict()
+
+        for ntype, node in chosen_nodes:
+            new_subg = remove_nodes(subg, node, ntype, store_ids=True)
+
+            if new_subg.num_edges() > 0:
+                new_subg_homo = to_homogeneous(new_subg)
+                # Get the largest weakly connected component in the subgraph.
+                nx_graph = to_networkx(new_subg_homo.cpu())
+                largest_cc_nids = list(
+                    max(nx.weakly_connected_components(nx_graph), key=len)
+                )
+                largest_cc_homo = node_subgraph(new_subg_homo, largest_cc_nids)
+                largest_cc_hetero = to_heterogeneous(
+                    largest_cc_homo, new_subg.ntypes, new_subg.etypes
+                )
+
+                # Follow steps for backtracking to original graph node ids
+                # 1. retrieve instanced homograph from connected-component homograph
+                # 2. retrieve instanced heterograph from instanced homograph
+                # 3. retrieve hetero-subgraph from instanced heterograph
+                # 4. retrieve orignal graph ids from subgraph node ids
+                cc_nodes = {
+                    ntype: subg.ndata[NID][ntype][
+                        new_subg.ndata[NID][ntype][
+                            new_subg_homo.ndata[NID][
+                                largest_cc_homo.ndata[NID][indicies]
+                            ]
+                        ]
+                    ]
+                    for ntype, indicies in largest_cc_hetero.ndata[NID].items()
+                }
+            else:
+                available_ntypes = [
```

Comment:
I assume `nx.weakly_connected_components` returns an empty list for a graph without edges?

Reply:
`nx.weakly_connected_components` is actually able to handle graphs without edges (returns sets containing single nodes), the branch here was due to an error in `dgl.to_homogeneous` when trying to convert from a heterogeneous graph with no edges